File: client/src/app/_models/pagination.ts
export class Pagination{
    page:number=1;
    itemsPerPage:number=10;
    totalItems:number;
    totalPages:number;

    constructor(){}

    castJsonToClass(json:Pagination){
        if(json.itemsPerPage){
          this.itemsPerPage = json.itemsPerPage;
        }
        if(json.page){
          this.page = json.page;
        }
        if(json.totalPages){
          this.totalPages = json.totalPages;
        }
        if(json.totalItems){
          this.totalItems = json.totalItems;
        }
      }
}

File: API/Helpers/Pagination.cs
namespace API.Helpers
{
    public class Pagination
    {
        private const int MaxPageSize=50;
        public int Page { get; set; }=1;
        public int TotalPages { get; set; }=1;
        public int ItemsPerPage { get; set; }=10;

    //         currentPage:number;
    // itemsPerPage:number;
    // totalItems:number;
    // totalPages:number;
        private int _pageSize=10;

        public int PageSize {
            get=>_pageSize;
            set=>_pageSize=(value>MaxPageSize)?MaxPageSize:value;
        }
    }
}